src/util/cron.c

```diff
@@ -187,19 +188,23 @@ static void *Cron_Run
 		// execute due tasks
 		CRON_TASK *task = NULL;
 		while((task = CRON_Peek()) && CRON_TaskDue(task)) {
+			pthread_mutex_lock(&cron->mutex);
 			CRON_TASK_STATE state = task->state;
 			// task state should be either pending or aborted
-			ASSERT(state == TASK_ABORT || state == TASK_PENDING);
+			ASSERT(state == TASK_ABORT || state == TASK_PENDING || state == TASK_INVALID);
 
 			// advance from pending to executing
 			if(CRON_TaskAdvanceState(task, TASK_PENDING, TASK_EXECUTING)) {
```

Comment:
in case all calls made to `CRON_TaskAdvanceState` are governed by  `pthread_mutex_lock(&cron->mutex);` then Task's state doesn't need to be `Atomic`